gateway/types/inforequest.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import (
+	"bytes"
+	"net/http"
+)
+
+// GatewayInfo provides information about the gateway and it's connected components
+type GatewayInfo struct {
+	Provider *ProviderInfo `json:"provider"`
+	Version  *VersionInfo  `json:"version"`
+}
+
+// ProviderInfo provides information about the configured provider
+type ProviderInfo struct {
+	Name          string       `json:"provider"`
+	Version       *VersionInfo `json:"version"`
+	Orchestration string       `json:"orchestration"`
+}
+
+// VersionInfo provides the commit message, sha and release version number
+type VersionInfo struct {
+	CommitMessage string `json:"commit_message,omitempty"`
+	SHA           string `json:"sha"`
+	Release       string `json:"release"`
+}
+
+// StringResponseWriter captures the handlers HTTP response in a buffer
+type StringResponseWriter struct {
+	body       *bytes.Buffer
+	headerCode int
+	header     http.Header
+}
+
+// NewStringResponseWriter create a new StringResponseWriter
```

Comment:
Maybe we could use the httptest recorder? There are some good examples in the ./handlers folder.

Reply:
Yes good idea